File: src/main/java/view/Rank.java
package view;

import javafx.beans.property.SimpleIntegerProperty;
import javafx.beans.property.SimpleStringProperty;

/**
 * The Rank class that allows to display the ranking data appropriately.
 */
public class Rank {
    private final SimpleStringProperty username;
    private final SimpleIntegerProperty points;

    public Rank(String user, Integer point) {
        this.username = new SimpleStringProperty(user);
        this.points = new SimpleIntegerProperty(point);
    }

    /**
     * Get the username of the person.
     *
     * @return
     */
    public String getUsername() {
        return username.get();
    }

    /**
     * Get the points of the person.
     *
     * @return
     */
    public Integer getPoints() {
        return points.get();
    }

}

File: src/main/scala/view/Rank.scala
package view

import javafx.beans.property.{SimpleIntegerProperty, SimpleStringProperty}
import javafx.collections.ObservableList

case class Rank (user: String, point: Int) {
  var username : SimpleStringProperty = new SimpleStringProperty(user)
  var points : SimpleIntegerProperty = new SimpleIntegerProperty(point)

  def getData: ObservableList[Rank] = ???
}
